### ISSUE TITLE
vue/no-dupe-keys: name match should be exact instead of indexOf

### ISSUE THREAD

**Issue by SomeoneIsWorking** [vuejs/eslint-plugin-vue#1738]
**What rule do you want to change?**
vue/no-dupe-keys
**Does this change cause the rule to produce more or fewer warnings?**
Fewer
**How will the change be implemented? (New option, new default behavior, etc.)?**
New default behavior
**Please provide some example code that this change will affect:**

```vue
export default {
  async asyncData() {
    const result = await getX();
    return {
      x = result
    }
  }
  data() {
    return {
      x = null
    }
  }
}
```

**What does the rule currently do for this code?**
Show "x" as duplicate key
**What will the rule do after it's changed?**
Not show "x" as duplicate key
**Additional context**
asyncData is part of a plugin I have
Currently I assume vue/no-dupe-keys thinks it is the same as data because of `usedNames.indexOf(o.name) !== -1`
But I actually need asyncData properties to be duplicated in data

**Comment by FloEdelmann** [comment]
Please see #1687, it is fixed already but not yet released.

**Comment by SomeoneIsWorking** [comment]
My bad. I only searched on google and didn't see a report on it before.
